hazelcast/src/main/java/com/hazelcast/internal/config/MapConfigReadOnly.java

```diff
@@ -161,6 +165,20 @@ private List<IndexConfig> getIndexConfigReadOnly() {
         return Collections.unmodifiableList(readOnlyIndexConfigs);
     }
 
+    private List<PartitioningAttributeConfig> getPartitioningAttributeConfigsReadOnly() {
+        List<PartitioningAttributeConfig> attributeConfigs = super.getPartitioningAttributeConfigs();
+        if (CollectionUtil.isEmpty(attributeConfigs)) {
+            return Collections.emptyList();
+        }
+
+        List<PartitioningAttributeConfig> readOnlyAttributeConfigs = new ArrayList<>(attributeConfigs.size());
+        for (PartitioningAttributeConfig attributeConfig : attributeConfigs) {
+            readOnlyAttributeConfigs.add(new PartitioningAttributeConfig(attributeConfig));
```

Comment:
I think we should have `PartitioningAttributeConfigReadOnly` class that throws exceptions in setters like `IndexConfigReadOnly`